plugins/user_management_plugin.py

```diff
@@ -92,8 +92,26 @@ def is_visible(self):
 
     #overrides BaseView method to check permission for the menu link
     def is_accessible(self):
+        if current_user.is_authenticated:
+            self.can_delete = True
+            self.can_create = True
+            if not is_super_user(self):
+                self.can_create = False
+                self.can_delete = False
+
         return current_user.is_authenticated
 
+    #overrides ModelView method to check default model actions
+    def is_action_allowed(self, name):
+        self.can_delete = True
+        self.can_create = True
+        if not is_super_user(self):
+            self.can_delete = False
+            self.can_create = False
+            return False
```

Comment:
Always returns False if user is not a super user. Does this mean that a user that is not a superuser can't edit their own profile?

Reply:
is_action_allowed hook is called before rendering the UI. This will block only the Create and Delete action icons being displayed in the list and other screens. Edit action is not blocked as the self.can_edit is always set to True.

Reply:
ok great thanks for confirming